dlk/python/dlk/templates/src/pack_input_to_qwords.cpp

```diff
@@ -24,99 +24,97 @@ limitations under the License.
-    return 0;
+    return;
   }
 #endif
 
 #ifdef USE_AVX
-  constexpr std::size_t SIMD_WIDTH = 32;
-  if ((input_depth % SIMD_WIDTH) == 0) {
-    const auto blocks = len / SIMD_WIDTH;
-    for (int i = 0; i < blocks; ++i) {
-      const auto a = _mm256_loadu_si256(reinterpret_cast<__m256i*>(input + i * SIMD_WIDTH));
-      const auto l = _mm256_movemask_epi8(_mm256_slli_epi16(a, 7));
-      const auto m = _mm256_movemask_epi8(_mm256_slli_epi16(a, 6));
-      output[i*2 + 0] = QUANTIZED_PACKED(l);
-      output[i*2 + 1] = QUANTIZED_PACKED(m);
+  if (bits_per_input == 2 && input_depth % bits_per_word == 0) {
+#pragma omp parallel for
+    for (size_t i = 0; i < area; ++i) {
+      for (size_t j = 0; j < full_words_in_depth; ++j) {
+        const auto a = _mm256_loadu_si256(reinterpret_cast<__m256i*>(input + i*blocks_in_depth*bits_per_word + j*bits_per_word));
+        const auto l = _mm256_movemask_epi8(_mm256_slli_epi16(a, 7));
+        const auto m = _mm256_movemask_epi8(_mm256_slli_epi16(a, 6));
+        output[i*bits_per_input + j*area*bits_per_input + 0] = QUANTIZED_PACKED(l);
+        output[i*bits_per_input + j*area*bits_per_input + 1] = QUANTIZED_PACKED(m);
+      }
     }
     Measurement::Stop();
-    return 0;
+    return;
   }
 #endif
 
-  for (int h = 0; h < input_height; ++h)
-      for (int w = 0; w < input_width; ++w) {
-          for (int d = 0; d < full_words_in_depth; ++d) {
-              output[current_word] = QUANTIZED_PACKED(0);
-              output[current_word + 1] = QUANTIZED_PACKED(0);
-              for (int d = 0; d < bits_per_word; ++d) {
-                for(int b = 0; b < bits_per_input; ++b)
-                  output[current_word + b] = QUANTIZED_PACKED(output[current_word + b].Raw() | ((input[input_index] >> b) & 1) << d);
-                input_index++;
-              }
-              current_word += bits_per_input;
-          }
+  for (size_t i = 0; i < area; ++i) {
+    for (size_t j = 0; j < full_words_in_depth; ++j) {
+      for(size_t b = 0; b < bits_per_input; ++b) {
+        QUANTIZED_PACKED tmp(0);
+        for (size_t d = 0; d < bits_per_word; ++d) {
+          QUANTIZED_PACKED::base_t in = input[i*input_depth + j*bits_per_word + d];
+          tmp |= QUANTIZED_PACKED(((in >> b) & 1) << d);
+        }
+        output[i*bits_per_input + j*area*bits_per_input + b] = tmp;
+      }
+    }
 
-          if(!remainder_bits_in_depth)
-              continue;
+    if(!remainder_bits_in_depth)
+      continue;
 
-          output[current_word] = QUANTIZED_PACKED(0);
-          output[current_word + 1] = QUANTIZED_PACKED(0);
-          for (int d = 0; d < remainder_bits_in_depth; ++d) {
-             for(int b = 0; b < bits_per_input; ++b)
-               output[current_word + b] = QUANTIZED_PACKED(output[current_word + b].Raw() | ((input[input_index] >> b) & 1) << d);
-             input_index++;
-          }
-          current_word += bits_per_input;
+    for(size_t b = 0; b < bits_per_input; ++b) {
```

Comment:
I'm not confident, but could you insert a space after `for` in this line? (I'm not confident because this line apparently came from old line 110, and the old line doesn't include a space. Should we change the code style when we just change indent, is very difficult problem for me. However, here, I'd like to have a space after `for` to keep consistency with the line 107.)

Reply:
Fixed.